### ISSUE TITLE
LB: load-balancer.hetzner.cloud/node-selector annotation not working

### ISSUE THREAD

**Issue by ydixken** [hetznercloud/hcloud-cloud-controller-manager#633]
### TL;DR

Hi there! We're creating a loadbalancer through the controller manager and our nodes are labeled within hcloud with `role=worker` - on service creation we see that the LB is set up correctly, but is missing all its targets. We currently have this set: 

```
        load-balancer.hetzner.cloud/node-selector: "role=worker"
```

We are able to configure the LB manually using the label, so there's no naming issue there, using the annotation it is not working. 

Is there any documentation for `node-selector` available?

### Expected behavior

We would expect that the node-selector annotation is working with the labels given. 

### Observed behavior

It does not :-) 

### Minimal working example

```# https://github.com/bitnami/charts/tree/main/bitnami/nginx-ingress-controller/#parameters
nginx_ingress_controller:
  chartVersion: 9.9.2
  values:
    kind: DaemonSet
    extraArgs:
      default-ssl-certificate: cert-manager/{{ .Values.baseClusterDomain }}
    ingressClassResource:
      name: nginx-ingress-controller
    service:
      annotations:
        load-balancer.hetzner.cloud/location: fsn1
        load-balancer.hetzner.cloud/name: lb01.k3s-ssi
        load-balancer.hetzner.cloud/use-private-ip: "true"
        load-balancer.hetzner.cloud/uses-proxyprotocol: "false"
        load-balancer.hetzner.cloud/http-redirect-https: "true"
        load-balancer.hetzner.cloud/node-selector: "role=worker"
        load-balancer.hetzner.cloud/disable-private-ingress: "true"
        load-balancer.hetzner.cloud/ipv6-disabled: "true"
      externalTrafficPolicy: Local
      type: LoadBalancer
      ports:
        http: 80
        https: 443
    publishService:
      enabled: true
    metrics:
      # Exposes Prometheus nginx_exporter at port 9913
      enabled: true
      serviceMonitor:
        enabled: true
        namespace: kube-prometheus-stack
```

**Comment by ydixken** [comment]
I can happily provide any logs, if needed. I watched out for the "unable to parse the node-selector annotation" in the log, according to https://github.com/hetznercloud/hcloud-cloud-controller-manager/blob/main/hcloud/load_balancers.go#L56 - but nothing there yet.

**Comment by apricote** [comment]
Hey @ydixken,

the labels need to be set on the Kubernetes Node objects.

**Comment by ydixken** [comment]
Makes sense, thank you! Closing.

**Comment by zap23** [comment]
Hi there,

I have been trying to get the feature described in this question to work, but I am having some problems. Can you confirm that it actually works? I followed the instructions, but I can't seem to get it to work. Any guidance would be greatly appreciated.

Thank you very much.

**Comment by apricote** [comment]
Hey @zap23,


Yes, this has worked before. Please open a new issue and describe the steps you took until you encountered the bug.
